src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -1249,6 +1345,57 @@ std::string ImageReplayer<I>::to_string(const State state) {
 }
 
 template <typename I>
+void ImageReplayer<I>::resync_image(Context *on_finish) {
+  dout(20) << dendl;
+
+  bool is_stopped = false;
+  {
+    Mutex::Locker l(m_lock);
+
+    m_stopping_for_resync = true;
+    if (m_state == STATE_STOPPED) {
```

Comment:
Yes, this is just to service the asok method. To use the librbd's `mirror_image_resync` when the replayer is stopped I still need to open the local image. If you think is still worth to use librbd's request resync method it's also fine by me.